File: lib/data/audio_playing_options.dart
import 'package:flutter_gen/gen_l10n/app_localizations.dart';
import 'package:rhasspy_mobile/data/option.dart';

class AudioPlayingOptions extends Option<AudioPlayingOption> {
  static final AudioPlayingOptions _singleton = AudioPlayingOptions._internal();

  factory AudioPlayingOptions() {
    return _singleton;
  }

  AudioPlayingOptions._internal() : super(AudioPlayingOption.values, AudioPlayingOption.disabled);

  @override
  String asText(AudioPlayingOption option, AppLocalizations local) {
    switch (option) {
      case AudioPlayingOption.local:
        return local.local;
      case AudioPlayingOption.remoteHTTP:
        return local.remoteHTTP;
      case AudioPlayingOption.remoteMQTT:
        return local.remoteMQTT;
      case AudioPlayingOption.disabled:
        return local.disabled;
    }
  }
}

enum AudioPlayingOption { local, remoteHTTP, remoteMQTT, disabled }

File: MultiPlatformLibrary/src/androidMain/kotlin/org/rhasspy/mobile/data/AudioPlayingOptions.kt
package org.rhasspy.mobile.data

import dev.icerock.moko.resources.StringResource
import org.rhasspy.mobile.MR

enum class AudioPlayingOptions(override val text: StringResource) : DataEnum {
    Local(MR.strings.local),
    RemoteHTTP(MR.strings.remoteHTTP),
    RemoteMQTT(MR.strings.remoteMQTT),
    Disabled(MR.strings.disabled)
}